r1cs/gadgets/std/src/pairing/mnt4/mod.rs

```diff
@@ -120,20 +122,23 @@ impl<P: MNT4Parameters> PairingGadget<MNT4p<P>, P::Fp> for MNT4PairingGadget<P>
         let elt = value.clone()
             .frobenius_map(cs.ns(|| "value_frobenius_2"), 2)?
             .mul(cs.ns(|| "value_frobenius_2_div_value"), &value_inv)?;
-        let elt_inv = value_inv.clone()
-            .frobenius_map(cs.ns(|| "value_inv_frobenius_2"), 2)?
-            .mul(cs.ns(|| "value_inv_frobenius_2_div_value"), &value)?;
 
         //Final exp last chunk
-
         let elt_q = elt.clone()
             .frobenius_map(cs.ns(|| "elt_q_frobenius_1"), 1)?;
 
         let w1_part = elt_q
             .cyclotomic_exp(cs.ns(|| "compute w1"), P::FINAL_EXPONENT_LAST_CHUNK_1)?;
 
-        let w0_part = elt_inv.clone()
-            .cyclotomic_exp(cs.ns(|| "compute w0"),P::FINAL_EXPONENT_LAST_CHUNK_ABS_OF_W0)?;
+        let w0_part = if P::FINAL_EXPONENT_LAST_CHUNK_W0_IS_NEG {
+            // we need the inverse of elt in this case
+            let elt_inv = value_inv.clone()
```

Comment:
Can you check if the .clone() is needed here ?

Reply:
It's not needed. I checked that via cargo test and pushed the commit.